tests/conftest.py

```diff
@@ -59,61 +63,134 @@ def cmd_line_args(request):
-        "address": "0x05b17e37fd43c18741877fca80846ad8c84aa750",
+        "api_port": 19093,
+        "p2p_port": 13303,
+        "api_token": DEFAULT_API_TOKEN,
+        "dir": f"{FIXTURE_FILES_DIR}{NODE_NAME_PREFIX}_3",
+        "host_addr": "localhost",
     },
     "4": {
-        "p2p_port": 19094,
-        "api_port": 13304,
-        "peer_id": "12D3KooWA494BRhXs2DpMm5e2DWkPZcot3WpYwB4KBj2udP9xvPC",
-        "address": "0xcc70a22331998454160472f097acb43ca9b1e646",
+        "api_port": 19094,
+        "p2p_port": 13304,
+        "api_token": DEFAULT_API_TOKEN,
+        "dir": f"{FIXTURE_FILES_DIR}{NODE_NAME_PREFIX}_4",
+        "host_addr": "127.0.0.1",
     },
     "5": {
-        "p2p_port": 19095,
-        "api_port": 13305,
-        "peer_id": "12D3KooWQCHVYhdnLT76rhHoFUu3b4L2aUiY8o1y8erhAwg8evFx",
-        "address": "0xe4bb1970e6c9e5689c5ef68ee2545b4366c49be4",
+        "api_port": 19095,
+        "p2p_port": 13305,
+        "api_token": DEFAULT_API_TOKEN,
+        "dir": f"{FIXTURE_FILES_DIR}{NODE_NAME_PREFIX}_5",
+        "host_addr": "localhost",
     },
     "6": {
-        "p2p_port": 19096,
-        "api_port": 13306,
-        "peer_id": "12D3KooWHc2LPyvYGLJbHoQeJUBXBjDRfMY7msobsPoWj8rCAGHr",
-        "address": "0xf90c1eb2557a443c2b27d399afac075fa752cd92",
+        "api_port": 19096,
+        "p2p_port": 13306,
+        "api_token": DEFAULT_API_TOKEN,
+        "dir": f"{FIXTURE_FILES_DIR}{NODE_NAME_PREFIX}_6",
+        "host_addr": "127.0.0.1",
+        "network": ANVIL_NETWORK2,
     },
     "7": {
-        "p2p_port": 19097,
-        "api_port": 13307,
-        "peer_id": "12D3KooWGPUAcSaJhKBmmwP2Bz3PbZyrErosBmCbZKgPPt7XHwqh",
-        "address": "0xe63ececd80c503548516e9e23ebb44d95c4d5ac2",
+        "api_port": 19097,
+        "p2p_port": 13307,
+        "api_token": DEFAULT_API_TOKEN,
+        "dir": f"{FIXTURE_FILES_DIR}{NODE_NAME_PREFIX}_7",
+        "host_addr": "localhost",
     },
 }
 
 
-def setup_node(*args, **kwargs):
-    logging.info(f"Setting up a node with configuration: {args} and {kwargs}")
-    pass
+def cleanup_node(args):
+    try:
+        proc = args["proc"]
+        proc.kill()
+    except:
+        pass
+
+
+def setup_node(args):
+    logging.info(f"Setting up a node with configuration: {args}")
+    log_file = open(f"{args['dir']}.log", "w")
+    network = args["network"] if "network" in args else ANVIL_NETWORK1
+    api_token = args["api_token"] if "api_token" in args else ""
+    api_token_param = f"--api-token={api_token}" if "api_token" in args else "--disableApiAuthentication"
+    custom_env = {
+        "DEBUG": "hopr*",
+        "NODE_ENV": "development",
+        "HOPRD_HEARTBEAT_INTERVAL": "2500",
+        "HOPRD_HEARTBEAT_THRESHOLD": "2500",
+        "HOPRD_HEARTBEAT_VARIANCE": "1000",
+        "HOPRD_NETWORK_QUALITY_THRESHOLD": "0.3",
+        "NODE_OPTIONS": "--experimental-wasm-modules",
+    }
+    cmd = [
+        "node",
+        "packages/hoprd/lib/main.cjs",
+        "--announce",
```

Comment:
Shouldn't we rather start using the YAML config file finally? At least for the non-variable portion. Would also allow testing with more complex strategy setups.

Reply:
One of the nodes already uses the YAML config. The difference with the current setup is that if the default values in YAML config change, they have to be manually updated in the test fixtures. Whereas with the command line approach, the valid defaults are always taken for values not specified in the command line.